### ISSUE TITLE
Direct struct interpretation sample

### ISSUE THREAD

**Issue by nixxholas** [lquerel/gcp-bigquery-client#15]
Could be good if we showcase an example of how we can deserialise the rows into a sturct via serde or avro!

https://github.com/nixxholas/gcp-bigquery-client/blob/4c30e1ca25281f67c5de08902a8a360974ea8c65/examples/client.rs#L198

**Comment by MikhailMS** [comment]
@nixxholas Are there currently any workaround to convert (deserialise) TableRow into a Struct via serde or maybe even directly into avro?
Looking at the code, it is not yet available, so curious about what options are there (except for hard-coding everything, which isn't good enough for my use-case)

**Comment by rahul-soshte** [comment]
Yeah examples regarding the data formats will be great. wanted something like arrow support as well. Actually the problem is when we make this examples public, the interoperability of this library with other libraries like pola-rs or maybe arrow becomes problematic.

**Comment by krystianity** [comment]
See -> https://github.com/lquerel/gcp-bigquery-client/issues/59#issuecomment-2119258440
